renku/core/metadata/repository.py

```diff
@@ -518,10 +579,69 @@ def get_global_configuration(writable: bool = False) -> "Configuration":
         return Configuration(repository=None, writable=writable)
 
     @staticmethod
-    def hash_object(path: Union[Path, str]) -> str:
+    def list_paths_in_revision(
+        paths: List[Union[Path, str]] = None, revision: str = "HEAD", working_dir="."
+    ) -> List[str]:
+        """List all paths that exist in a revision.
+
+        Warning: ls-tree is sensitive to the current working directory, make sure to set working_dir to repo root.
```

Comment:
It's perhaps cleaner to calculate relative paths to the working_dir in. this function instead of relying on the caller to pass the correct values.